app/javascript/app/pages/custom-compare/custom-compare-selectors.js

```diff
@@ -17,42 +27,39 @@ const getCountryOptions = createSelector([getCountries], countries => {
   }));
 });
 
-const getSelectedCountries = createSelector(
-  [getCountryOptions, getQuery],
-  (countriesData, query) => {
-    if (!countriesData && !countriesData.length && !query) return null;
-    const { country0, country1, country2 } = query;
-    return {
-      country0,
-      country1,
-      country2
-    };
-  }
-);
-
-const getSelectedDocuments = createSelector([getQuery], query => {
+export const getSelectedTargets = createSelector([getQuery], query => {
   if (!query) return null;
-  const { document0, document1, document2 } = query;
-  return {
-    document0,
-    document1,
-    document2
-  };
+  const queryTargets = query.targets ? query.targets.split(',') : [];
+  return [1, 2, 3].map((value, i) => {
+    // targets are saved as a string 'ISO3-DOCUMENT', e.g. 'USA-NDC'
+    const target =
+      queryTargets && queryTargets[i] && queryTargets[i].split('-');
+    const country = target && target.length > 0 && target[0];
```

Comment:
Why this extra length > 0? And shouldn't it be only length > 0 for document?

Reply:
:man_facepalming: Oh, you're right,  we don't need to check `target.length` in both cases, we will get `undefined` anyway, thanks!